routing/payment_lifecycle.go

```diff
@@ -258,8 +257,8 @@ lifecycle:
 
 		// Create a new payment attempt from the given payment session.
 		rt, err := p.paySession.RequestRoute(
```

Comment:
Good suggestion. I think we should modify `RequestRoute` to temporarily "lock" the amount to avoid the liquidity racing.

Reply:
what do you mean by saying "real max amount"? I think the `max_amount` is the `remainingAmt` here and we update it for every iteration?

Reply:
So when the user restricts the max_amount with --max_shard_size_msat we will only check in our pathfinding whether we do have enough liquidity overall to send the new max_size, but in my view we should always check the real max_amount (https://github.com/lightningnetwork/lnd/blob/master/routing/pathfind.go#L521). So that we do not sent a payment out although we would have not enough to pay the whole amount ?